fix(components):[select] placeholder local

gjfei opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

gjfei committed:
fix(components):[select] placeholder local

kooriookami committed:
fix(docs): fix Cross hovering (#7558)

YunYouJun committed:
fix(components): input internal autofill border color (#7561)

JobinJia committed:
fix(components): [el-tree-v2]prop `icon` support Component. fix [#7517] (#7518)

* fix(components): [el-tree-v2]prop `icon` support Component. fix [#7517]

* fix(components): [el-tree-v2]prop `icon` support Component. fix [#7517]

0song committed:
chore: add right sponsors (#7567)

* chore: add spon

* chore: add spon dark

* chore: add spon ui

* chore: add spon ui2

* chore: add analytics

* chore: add readme

chenxch committed:
fix: tree-select nodeKey default selected (#7172) (#7181)

Fix the problem that nodeKey initialization is invalid

anflower committed:
fix(component): cascader searching placeholder does not disappear (#7540)

0song committed:
chore: update slogan text (#7573)

YunYouJun committed:
fix(components): select v2 remote method trigger reset value when blur (#7569)

JeremyWuuuuu commented on packages/components/select/src/select.vue:
```diff
@@ -513,16 +514,25 @@ export default defineComponent({
-        props.modelValue.length > 0
-      ) {
-        currentPlaceholder.value = ''
+    watch(
+      lang,
+      () => {
+        states.cachedPlaceHolder = currentPlaceholder.value =
+          props.placeholder || t('el.select.placeholder')
+        if (
+          props.multiple &&
+          Array.isArray(props.modelValue) &&
+          props.modelValue.length > 0
+        ) {
+          currentPlaceholder.value = ''
+        }
+      },
+      {
+        immediate: true,
       }
+    )
```
**Suggestion:** put this chunk of code inside `onMounted`

gjfei replied:
> **Suggestion:** put this chunk of code inside `onMounted`

可以告诉我原因吗？

JeremyWuuuuu replied:
No specific reason, we should keep the change minimal. 

If you think the watch statement is ok to be put outside `onMounted` hook, you can make another PR for that.

Does this make any sense?

gjfei replied:
我好像不是很能理解watch放在onMounted里面，如果是用工具去对比变化的话，并没有过多的更改
![image](https://user-images.githubusercontent.com/33827314/167333621-0e33ea9f-d6f2-4d72-a6e4-8aeabd272c21.png)

JeremyWuuuuu replied:
If you insist, would you mind adding a test case against your change?

jw-foss reviewed (changes requested):
Left a comment for you to consider 🙇

element-bot committed:
chore: update changelog 2.2.0 (#7579)

ryuhangyeong committed:
fix(components): circle progress 0 percent transition effect (#7562)

YunYouJun committed:
fix(components): date picker range input__inner inherit height (#7586)

YunYouJun committed:
fix(components): input textarea padding & label line-height (#7570)

* fix(components): input textarea padding & label line-height

* fix(components): form label top align by line-height

jw-foss committed:
improvement(project): breaking change in yml (#7593)

- Add breaking change yaml example.
- Add breaking change list for 2.2.

YunYouJun committed:
ci: simplify pr-docs-build & remove macos-build (#7597)

YunYouJun committed:
docs: add how to custom namespace & example (#7596)

btea committed:
fix(docs): copy icon (#7587)

* fix(docs): copy icon

* chore: modify demo & update copy

plainheart committed:
fix(theme-chalk): [input] fix unexpected var name of input-inner-height (#7600)

zouhangwithsweet committed:
chore: support gitpod

zouhangwithsweet committed:
chore: update yml

YunYouJun committed:
fix: datepicker range default height by map.get (#7608)

satrong committed:
fix(theme-chalk): [input] add not-allowed when disabled (#7612)

opengraphica committed:
feat(components): [dialog] a11y and focus trap on dialog components (#7550)

* feat(components): [dialog] a11y and focus trap on dialog components

* feat(components): [dialog] remove default browser focus style

* feat(components): [dialog] clean up async and comment

YunYouJun committed:
ci: add strategy & use setup-node cache for workflows (#7621)

opengraphica committed:
feat(components): [date-picker] work with modal focus trap; a11y controls and attributes (#7598)

* feat(components): [date-picker] a11y controls and attributes

* feat(components): [date-picker] keyboard controls for picker

* feat(components): [date-picker] unit test complete

* feat(components): [date-picker] remove immediate watch date

jw-foss committed:
feat(ci): enforce commit message linter (#7609)

- Enforce lint commit message with conventional commit.

chenxch committed:
fix(components): [select] in form change size (#7599)

* fix: select in form change size

* fix: unuser

jw-foss committed:
fix(ci): replace token for lint notification (#7625)

- Replace secrets.GITHUB_TOKEN with github.token

chenxch committed:
fix: check-strictly bug (#7422)

iamkun committed:
chore: update changelog (#7640)

tolking committed:
chore(ci): optimizing the generated code hint file (#7634)

jw-foss committed:
fix(ci): token limitation for commit lint (#7647)

- Update gitlab.token to BOT_TOKEN for managing triage thing.

iamkun committed:
chore: update index spon (#7663)

CarterLi committed:
feat(components): menu add more controllable styles (#7549)

1. add some css variables
2. support specifying level padding
3. small code refactor
4. deprecate `background-color` etc

tolking committed:
docs: only cache the user preferred language pages in the PWA (#7568)

* docs: cache the preferred lang pages in the PWA

* chore: remove index.html in manifest

* fix: no cache the '/' page

* perf: optimize code

* docs: add always refresh from PWA

iamkun committed:
chore: doc temp remove PWA to fix server error (#7666)

iamkun committed:
chore: fix sw.js error (#7667)

ryuhangyeong committed:
fix(components): tabs focus visible (#7665)

jw-foss committed:
fix(ci): token limitation for commit lint (#7675)

- Set lint result to an artifact.
- Add a workflow which is responsible for posting.

sxzz committed:
refactor(build): use type unsafe (#7523)

* refactor(build): use type unsafe

* fix: windows

iamkun committed:
chore: enable mirror (#7688)

jw-foss committed:
fix(ci): post lint message (#7692)

- Add context for download artifact

chenxch committed:
feat(components): [button] add link api (#7652)

* feat: button add link type

* fix: code lint

* feat: button add link api

* docs: edit word

Co-authored-by: xiaochenchen <xiaochen.chen@igg.com>

jw-foss committed:
docs(components): [button] link button (#7694)

- Update docs for link button.
- Update vp-translation wrong usage of button.
- Update doc show cases examples.
- Update commit hash value for `link button`

jw-foss committed:
fix(ci): post lint message (#7695)

- Fix post lint message with dawidd6's actions

ryuhangyeong committed:
improvement(components) [divider] add separator role (#7678)

ryuhangyeong committed:
fix(components): [upload] show close tip text (#7696)

YunYouJun committed:
fix(components): [date-picker] date-table-cell namespace (#7684)

* fix(components): date-table-cell & scrollbar namespace

* fix(components): revert scroll className & add comment

jw-foss committed:
improvement(theme-chalk): visual enhancement (#7563)

* improvement(theme-chalk): visual enhancement

- Add visual enhancement for keyboard nagivation on form items.

* Fix linter issue

* Fix switch active text issue

* Fix bordered radio demo issue

lily-elephant committed:
fix(components): date-picker add keydown events(#7506) (#7536)

* fix(components): date-picker add keyup events(#7506)

* Update picker.vue

feat(components): keyup变更为keydown

* Update picker.vue

fix(components): picker delete superfluous code

* Update picker.vue

fix(components): emit事件

* feat: date-picker & timer-picker add unit test

* feat: picker add unit test expected results

* Update time-picker.test.ts

feat: 去除lint error

btea committed:
fix(components): [cascader] disabled option can't trigger cancel checked (#7604)

ryuhangyeong committed:
refactor(components): [main] switch to script-setup syntax (#7707)

* refactor(components): [main] switch to script-setup syntax

* refactor(components): [main] switch to script-setup syntax

metanas committed:
refactor(components): [timeline] switch to script-setup syntax (#7705)

Co-authored-by: metanas <matanas@pre-history.com>

metanas committed:
refactor(components): [collection] switch to script-setup syntax (#7710)

Co-authored-by: metanas <matanas@pre-history.com>

SignDawn committed:
Update basic.vue (#7722)

文档说的一千条，实际只有 200 条

CarterLi committed:
feat(components): [input-number] add :value-on-clear and make it nullable (#7724)

btea committed:
fix(components): [table-v2] don't spread headerClass (#7729)

jw-foss committed:
docs(components): [input-number] documentation updates (#7727)

- Add version tag for `value-on-clear`.
- Add breaking change log for `value-on-clear`.

jw-foss committed:
fix(ci): lint commit message error (#7731)

- Add PR numbers for workflow.

jw-foss committed:
feat(project): introducing triage bot (#7732)

- Add triage bot configuration.

jw-foss committed:
chore(ci): bump pr-helper version (#7734)

chenxch committed:
refactor(components): [radio] switch to script-setup synatx (#7701)

* refactor: radio script setup

* refactor: radio script setup

btea committed:
refactor(components): [container] switch to script-setup syntax (#7725)

* refactor(components): [container] switch to script-setup syntax

* refactor: remove uncessary default

purepear committed:
test: remove deprecated done callback (#7741)

Co-authored-by: 三咲智子 <sxzz@sxzz.moe>

renovate[bot] committed:
chore(deps): update actions/upload-artifact action to v3 (#7753)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update all non-major dependencies (#7754)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: 三咲智子 <sxzz@sxzz.moe>

chenxch committed:
docs: remove radio in type-unsafe (#7757)

metanas committed:
docs(type-unsafe): [pagination] remove (#7759)

Co-authored-by: metanas <matanas@pre-history.com>

jw-foss committed:
feat(components): [button] restore type="text" (#7761)

metanas committed:
refactor(components): [pagination/pager] switch to script-setup syntax (#7746)

* refactor(components): [pagination/pager] switch to script-setup syntax

* fix(components): [pagination/pager] cast props to const

Co-authored-by: metanas <matanas@pre-history.com>

metanas committed:
refactor(components): [pagination/jumper] switch to script-setup syntax (#7747)

Co-authored-by: metanas <matanas@pre-history.com>

metanas committed:
refactor(components): [pagination/total] switch to script-setup syntax (#7750)

* refactor(components): [pagination/total] switch to script-setup syntax

* fix(components): [pagination/total] cast props to const

Co-authored-by: metanas <matanas@pre-history.com>

buqiyuan committed:
fix(components): [form-item] label is not vertically centered (#7780)

YunYouJun committed:
fix(theme-chalk): [DatetimePicker] change inner to wrapper & use css var (#7703)

* fix(theme-chalk): [DatetimePicker] change inner to wrapper & use css var for width

* test: use wrapper to fix test

metanas committed:
refactor(components): [pagination/prev] switch to script-setup syntax (#7745)

* refactor(components): [pagination/prev] switch to script-setup syntax

* fix(components): [pagination/prev] cast props to const

* fix(components): [pagination/prev] extract emits to utility file

* fix(components): [pagination/prev] remove prop default

Co-authored-by: metanas <matanas@pre-history.com>

metanas committed:
refactor(components): [pagination/next] switch to script-setup syntax (#7748)

* refactor(components): [pagination/next] switch to script-setup syntax

* fix(components): [pagination/next] move emits

Co-authored-by: metanas <matanas@pre-history.com>

buqiyuan committed:
fix(components): [time-picker] can not hide panel when click frequently (#7779)

CarterLi committed:
fix(components): [popper] provide formItemContext conditionally (#7790)

0song committed:
docs: update doc (#7799)

btea committed:
fix(hooks): use-form-item circular reference (#7793)

chenxch committed:
refactor(components): [checkbox] switch to script-setup syntax (#7794)

* refactor: checkbox switch to script-setup syntax

* refactor: checkbox use useSizeProps and edit test

metanas committed:
refactor(components): [pagination/sizes] switch to script-setup syntax (#7749)

* refactor(components): [pagination/sizes] switch to script-setup syntax

* fix(components): [pagination/sizes] cast props to const

* fix(components): [pagination/sizes] remove prop default

Co-authored-by: metanas <matanas@pre-history.com>

jw-foss committed:
fix(components): [dialog/drawer] header slot

- Fix header slot rendering for both dialog and drawer.
- Change the deprecation for `title` slot to **3.0.0**.

sxzz committed:
fix: improve deps (#7763)

YunYouJun committed:
refactor(theme-chalk): [table] remove useless style (#7807)

msidolphin committed:
feat(components): [table] ensure minimum-size doesn't follow the content (#7627)

* feat(components): [table] ensure minimum-size doesn't follow the content

* chore: add version tag

opengraphica committed:
feat(components): [popper] focus trap and a11y (#7736)

* feat(components): [popper] fix focus traps

* feat(components): [popper] add focus trap tests

Co-authored-by: JeremyWuuuuu <15975785+JeremyWuuuuu@users.noreply.github.com>

ryuhangyeong committed:
refactor(components): [step] switch to script-setup syntax (#7782)

* refactor(components): [step] switch to script-setup syntax

* improvement(components): [step] props type

* improvement(components): [step] unnecessary type

ryuhangyeong committed:
refactor(components): [steps] switch to script-setup syntax (#7783)

* refactor(components): [steps] switch to script-setup syntax

improvement(components): [steps] props type

improvement(components): [steps] unnecessary type

* fix(comoponents): [steps] import error

* fix(comoponents): [steps] import error

chenxch committed:
fix(radio-group): restore name (#7810)

* fix(radio-group): restore name

* fix(radio-group): restore name

renovate[bot] committed:
chore(deps): update dependency husky to v8 (#7574)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update dependency puppeteer to v14 (#7756)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

element-bot committed:
chore: update changelog 2.2.1 (#7812)

tolking committed:
fix(components): [menu] the styles link error (#7820)

iamkun committed:
chore: enable verbose crowdin (#7817)

zhoucan38 committed:
feat(Dialog): expose dialogContentRef (#7633)

ryuhangyeong committed:
fix(components): [form] align label width if it exists (#7826)

0song committed:
fix: correct the deprecated note for el-button (#7854)

fix: correct the deprecated note for el-button

gjfei committed:
fix(components): [select] dynamic change placeholder local
